tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -283,6 +283,97 @@ def _make_same_in_out_dtypes(number_of_in_params, dtypes):
     _in_out_dtypes_float_array_float_scalar)
 
 
+_in_out_dtypes_integer_mixed = [
```

Comment:
How about renaming to `_in_out_dtypes_binary_math_functions` and moving this definition just after the definition of `_in_out_dtypes_math_functions`?

Reply:
Move only that, or the other lists like `_in_out_dtypes_float_mixed, _in_out_dtypes_float_integer_mixed, etc` as well?

Reply:
Yes. It is because this rule is different from the rule for arithmetics operations (e.g., `add`).
It seems better to merge these lists. How about writing as follows?
```py
_in_out_dtypes_math_functions = dtype_utils._permutate_dtype_mapping([
    # integers
    (('int8', 'int16'), 'float32'),
    ...
    # integer and float
    (('int8', 'float16'), 'float16'),
    ...
    # floats
    ('float16', 'float32'), 'float32'),
    ....
])

Reply:
Oh got it. On it.